server/cabal.project.dev-sh

```diff
@@ -0,0 +1 @@
+cabal.project
```

Comment:
What is the use case behind not just using the `cabal.project` file directly? I can’t imagine a situation where I would want `dev.sh` to use a different `cabal.project` file than the one that’s checked in.

Reply:
We symlink `cabal.project` to `cabal.project.dev-sh` since the only way to specify a new `cabal.project.local`  is to do so indirectly via specifying the cabal.project file name

Reply:
Ah, I see—the trick is that if you specify `--project-file=foobar`, then `cabal-install` looks for `foobar.local` instead of `cabal.project.local`. That makes sense.

I don’t like this proliferation of symlinks and duplicate configuration files—I think it’s confusing—but I don’t have a better solution right now. One option would be for `dev.sh` to manage its own temporary directory (via `mktemp -d`), where it could copy files or create symlinks and muck about with them as desired. I don’t know if that would be more complicated than you’d like, however.

IMO, the real right thing to do here is upstream some changes to `cabal-install`. I’d love to improve some of these things—it’s not clear to me that the behavior with things like `--enable-coverage` isn’t just a bug (IIUC flags are only *supposed* to affect local packages). But I recognize that in the meantime we need a workflow that works!

Reply:
> --it’s not clear to me that the behavior with things like --enable-coverage

ya I agree, I think I referenced a somewhat relevant issue in comments

> the real right thing to do here is upstream some changes to cabal-install

Right, but I think there's no way to avoid hacks and workarounds unless we're going to ship a cabal-install fork with graphql-engine. My hope is this script means just one person has to figure out these hacks and corner cases. It would be great to devote some resources to cabal-install though

Reply:
Yes, agreed. I’m alright with this trick for now, and maybe we can fix `cabal-install` in the future.